de.metas.acct.base/src/main/sql/postgresql/ddl/functions/AccountSheetReport.sql

```diff
@@ -0,0 +1,238 @@
+    DROP TABLE IF EXISTS TMP_AccountSheetReport;
+    CREATE TEMPORARY TABLE TMP_AccountSheetReport AS
+        (
+            SELECT fa.*,
+                   tc.c_taxcategory_id,
+                   0::numeric beginningBalance,
+                   0::numeric endingBalance,
+                   NULL::text lineType
+            FROM c_elementvalue ev
+                     INNER JOIN fact_acct fa ON ev.c_elementvalue_id = fa.account_id -- only search for accounts with at least 1 transaction
+                     LEFT JOIN c_tax t ON fa.c_tax_id = t.c_tax_id
+                     LEFT JOIN c_taxcategory tc ON t.c_taxcategory_id = tc.c_taxcategory_id
+            WHERE FALSE
+        );
+
+    v_time := logDebug('created empty temporary table', v_time);
+
+
+    --
+    -- insert into the temp table the beginningBalance for all the available accounts which have the balance != 0
+    -- noinspection SqlInsertValues
+    WITH filteredElementValues AS
+             (
+                 SELECT ev.c_elementvalue_id
+                 FROM c_elementvalue ev
+                 WHERE TRUE
+                   AND (p_account_id IS NULL OR ev.c_elementvalue_id = p_account_id)
+                 ORDER BY ev.c_elementvalue_id
+             ),
+         previousBalances AS
+             (
+                 SELECT (de_metas_acct.acctBalanceToDate(ev.c_elementvalue_id, p_c_acctschema_id, (p_dateFrom - INTERVAL '1 day')::date, p_ad_org_id)::de_metas_acct.BalanceAmt).Balance previousDayBalance,
+                        ev.c_elementvalue_id
+                 FROM filteredElementValues ev
+             ),
+         nonZeroPreviousBalances AS
+             (
+                 SELECT *
+                 FROM previousBalances
+                 WHERE previousDayBalance != 0
+             )
+    INSERT
+    INTO TMP_AccountSheetReport (beginningBalance, endingBalance, lineType, account_id)
+    SELECT nonZero.previousDayBalance,
+           nonZero.previousDayBalance,
+           LINE_TYPE_BEGINNINGBALANCE,
+           nonZero.c_elementvalue_id
+    FROM nonZeroPreviousBalances nonZero;
+
+    GET DIAGNOSTICS v_temp = ROW_COUNT;
+    v_time := logDebug('inserted beginningBalance: ' || v_temp || ' records', v_time);
+
+
+    --
+    -- insert the fact_acct rows into the table
+    WITH filteredFactAcct AS
+             (
+                 SELECT fa.*,
+                        tc.c_taxcategory_id,
+                        tmp_fa.beginningBalance::numeric beginningBalance,
+                        tmp_fa.endingBalance::numeric    endingBalance,
+                        LINE_TYPE_TRANSACTION
+                 FROM fact_acct fa
+                          INNER JOIN TMP_AccountSheetReport tmp_fa ON tmp_fa.account_id = fa.account_id --
+                          LEFT JOIN c_tax t ON fa.c_tax_id = t.c_tax_id
+                          LEFT JOIN c_taxcategory tc ON t.c_taxcategory_id = tc.c_taxcategory_id
```

Comment:
don't need this join

Reply:
which join? (from the comment/preview i don't understand to which join you're referring to)

Reply:
was talking about tax category..... but now u need it because u need tax category's name

Reply:
okay. marking as resolved